src/read.c

```diff
@@ -3607,198 +3559,267 @@ static avifProperty * avifDecoderItemAddProperty(avifDecoderItem * item, const a
-        AVIF_CHECKERR(iccDataSize != UINT32_MAX, AVIF_RESULT_BMFF_PARSE_FAILED);
+    uint32_t iccDataSize = 0;
+    if (hasIcc) {
+        AVIF_CHECKERR(avifROStreamReadBits(s, &iccDataSize, metadataSizeSize),
+                      AVIF_RESULT_BMFF_PARSE_FAILED); // unsigned int(metadata_size_size) icc_data_size_minus_one;
         ++iccDataSize;
-    } else if (colorType == AVIF_MINI_COLOR_TYPE_SRGB) {
-        // sRGB
-        colorPrimaries = AVIF_COLOR_PRIMARIES_SRGB;
-        transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB;
-        matrixCoefficients = isMonochrome ? AVIF_MATRIX_COEFFICIENTS_UNSPECIFIED : AVIF_MATRIX_COEFFICIENTS_BT601;
-        iccDataSize = 0;
+    }
+
+    uint8_t colorPrimaries;
+    uint8_t transferCharacteristics;
+    uint8_t matrixCoefficients;
+    if (hasExplicitCicp) {
+        AVIF_CHECKERR(avifROStreamReadBits8(s, &colorPrimaries, 8), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(8) colour_primaries;
+    } else if (hasIcc) {
+        colorPrimaries = AVIF_COLOR_PRIMARIES_UNSPECIFIED; // 2
     } else {
-        const uint32_t numBitsPerComponent = (colorType == AVIF_MINI_COLOR_TYPE_NCLX_5BIT) ? 5 : 8;
-        AVIF_CHECKERR(avifROStreamReadBits(&s, &colorPrimaries, numBitsPerComponent),
-                      AVIF_RESULT_BMFF_PARSE_FAILED); // bit(5/8) color_primaries;
-        AVIF_CHECKERR(avifROStreamReadBits(&s, &transferCharacteristics, numBitsPerComponent),
-                      AVIF_RESULT_BMFF_PARSE_FAILED); // bit(5/8) transfer_characteristics;
-        if (isMonochrome) {
-            matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_UNSPECIFIED;
-        } else {
-            AVIF_CHECKERR(avifROStreamReadBits(&s, &matrixCoefficients, numBitsPerComponent),
-                          AVIF_RESULT_BMFF_PARSE_FAILED); // bit(5/8) matrix_coefficients;
-        }
-        iccDataSize = 0;
+        colorPrimaries = AVIF_COLOR_PRIMARIES_SRGB; // 1
+    }
+    if (hasExplicitCicp) {
+        AVIF_CHECKERR(avifROStreamReadBits8(s, &transferCharacteristics, 8), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(8) transfer_characteristics;
+    } else if (hasIcc) {
+        transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_UNSPECIFIED; // 2
+    } else {
+        transferCharacteristics = AVIF_TRANSFER_CHARACTERISTICS_SRGB; // 13
+    }
+    if (chromaSubsampling == 0) {
+        // Monochrome
+        matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_UNSPECIFIED; // 2
+    } else if (hasExplicitCicp) {
+        AVIF_CHECKERR(avifROStreamReadBits8(s, &matrixCoefficients, 8), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(8) matrix_coefficients;
+    } else {
+        matrixCoefficients = AVIF_MATRIX_COEFFICIENTS_BT601; // 6
     }
 
-    uint32_t hasExplicitCodecTypes;
-    AVIF_CHECKERR(avifROStreamReadBits(&s, &hasExplicitCodecTypes, 1), AVIF_RESULT_BMFF_PARSE_FAILED); // bit(1) has_explicit_codec_types;
+    char infeType[] = "av01";
+    char codecConfigType[] = "av1C";
```

Comment:
Declare these as `uint8_t` arrays of size 4 and delete the `(uint8_t *)` casts in line 3645 and line 3646.

See the `type` field in the `avifDecoderItem` struct:
```
typedef struct avifDecoderItem
{
    ...
    uint8_t type[4];
    ...
} avifDecoderItem;
```

The default values can be set using `memcpy()` calls. Also, the default values can be set in a new `else` branch at line 3654, similar to how we set the default values for `colorPrimaries`, `transferCharacteristics`, etc.